src/common/heartbeat/src/listener.cpp

```diff
@@ -0,0 +1,26 @@
+#include <rclcpp/rclcpp.hpp>
+#include <std_msgs/msg/bool.hpp>
+
+#include <iostream>
+
+class Listener : public rclcpp::Node {
+public:
+    Listener() : Node("heartbeat_lsitener") {
+        using namespace std::chrono_literals;
+        heartbeat_subscription = this->create_subscription<std_msgs::msg::Bool>("heartbeat", 10, std::bind(&Listener::heartbeat, this, std::placeholders::_1));
```

Comment:
Any particular reason for switching this to C++? Besides, you know, C++ masochism?